packages/build-utils/src/lambda.ts

```diff
@@ -12,19 +12,27 @@ interface Environment {
   [key: string]: string;
 }
 
-export interface LambdaOptions {
-  files: Files;
+export type LambdaOptions = LambdaOptionsWithFiles | LambdaOptionsWithZipBuffer;
+
+export interface LambdaOptionsBase {
   handler: string;
   runtime: string;
   memory?: number;
   maxDuration?: number;
   environment?: Environment;
   allowQuery?: string[];
   regions?: string[];
+}
+
+export interface LambdaOptionsWithFiles extends LambdaOptionsBase {
+  files: Files;
+}
+
+export interface LambdaOptionsWithZipBuffer extends LambdaOptionsBase {
   /**
    * @deprecated Use `files` property instead.
    */
```

Comment:
Should we move the `LambdaOptionsWithZipBuffer` to be deprecated instead of the property?

Reply:
Added.